code/parachain/frame/composable-traits/src/currency.rs

```diff
@@ -21,6 +21,12 @@ pub trait CurrencyFactory<AssetId, Balance> {
 	fn reserve_lp_token_id(ed: Balance) -> Result<AssetId, DispatchError> {
 		Self::create(RangeId::LP_TOKENS, ed)
 	}
+	/// Given a local `AssetId` (withen the range of `0` to `u32::MAX`) returns a global (within the
+	/// scope of Currency Factory) `AssetId`.
+	fn local_to_global_asset_id(
+		asset_id: AssetId,
```

Comment:
would it make sense to have the asset_id parameter be a u32?

Reply:
It's easier to safely go from `u32` to `u128` than the other way around, and we don't know what context this may be called in.
So, maybe, but doing it this way does prevent the need for more error checking elsewhere.

I could go either way on this.

Reply:
I guess it depends on what exactly the asset_id param is, is it something returned from another function in this pallet?

i.e. when would this be called, and with what values

Reply:
going to resolve this, it works and can be discussed later